src/ADAL.PCL.Mac/AuthenticationAgentNSWindowController.cs

```diff
@@ -0,0 +1,284 @@
+            }
+
+            // Calculate the center of the current main window so we can position our window in the center of it
+            CGRect centerRect = CenterRect(windowRect, new CGRect(0, 0, DEFAULT_WINDOW_WIDTH, DEFAULT_WINDOW_HEIGHT));
+
+            var window = new NSWindow(centerRect, NSWindowStyle.Titled | NSWindowStyle.Closable | NSWindowStyle.Resizable, NSBackingStore.Buffered, true)
+            {
+                BackgroundColor = NSColor.Red,
+                WeakDelegate = this,
+                AccessibilityIdentifier = "ADAL_SIGN_IN_WINDOW"
+            };
+
+            var contentView = window.ContentView;
+            contentView.AutoresizesSubviews = true;
+
+            webView = new WebView(contentView.Frame, null, null)
+            {
+                FrameLoadDelegate = this,
+                PolicyDelegate = this,
+                AutoresizingMask = NSViewResizingMask.HeightSizable | NSViewResizingMask.WidthSizable,
+                AccessibilityIdentifier = "ADAL_SIGN_IN_WEBVIEW"
+            };
+
+            contentView.AddSubview(webView);
+
+            progressIndicator = new NSProgressIndicator(new CGRect(DEFAULT_WINDOW_WIDTH / 2 - 16, DEFAULT_WINDOW_HEIGHT / 2 - 16, 32, 32))
+            {
+                Style = NSProgressIndicatorStyle.Spinning,
+                // Keep the item centered in the window even if it's resized.
+                AutoresizingMask = NSViewResizingMask.MinXMargin | NSViewResizingMask.MaxXMargin | NSViewResizingMask.MinYMargin | NSViewResizingMask.MaxYMargin
+            };
+
+            // On OS X there's a noticable lag between the window showing and the page loading, so starting with the spinner
+            // at least make it looks like something is happening.
+            progressIndicator.Hidden = false;
+            progressIndicator.StartAnimation(null);
+
+            contentView.AddSubview(progressIndicator);
+
+            Window = window;
+
+            webView.MainFrameUrl = url;
+        }
+
+        static CGRect CenterRect(CGRect rect1, CGRect rect2)
+        {
+            nfloat x = rect1.X + ((rect1.Width - rect2.Width) / 2);
+            nfloat y = rect1.Y + ((rect1.Height - rect2.Height) / 2);
+
+            x = x < 0 ? 0 : x;
+            y = y < 0 ? 0 : y;
+
+            rect2.X = x;
+            rect2.X = y;
+
+            return rect2;
+        }
+
+        [Export("webView:decidePolicyForNavigationAction:request:frame:decisionListener:")]
+        void DecidePolicyForNavigation(WebView webView, NSDictionary actionInformation, NSUrlRequest request, WebFrame frame, NSObject decisionToken)
+        {
+            if (request == null)
+            {
+                WebView.DecideUse(decisionToken);
+                return;
+            }
+
+            string requestUrlString = request.Url.ToString();
+
+            if (requestUrlString.StartsWith(BrokerConstants.BrowserExtPrefix, StringComparison.OrdinalIgnoreCase))
```

Comment:
launching browser in Mac for device enrollment is not  a supported scenario. native objc library returns error with "server is redirecting us to browser, this behavior is not defined on Mac OS X yet" message